static/app/views/profiling/functions/content.tsx

```diff
@@ -0,0 +1,264 @@
+import {useCallback, useMemo} from 'react';
+import {browserHistory} from 'react-router';
+import styled from '@emotion/styled';
+
+import Count from 'sentry/components/count';
+import DatePageFilter from 'sentry/components/datePageFilter';
+import EnvironmentPageFilter from 'sentry/components/environmentPageFilter';
+import GridEditable, {
+  COL_WIDTH_UNDEFINED,
+  GridColumnOrder,
+} from 'sentry/components/gridEditable';
+import * as Layout from 'sentry/components/layouts/thirds';
+import PageFilterBar from 'sentry/components/organizations/pageFilterBar';
+import PerformanceDuration from 'sentry/components/performanceDuration';
+import SmartSearchBar, {SmartSearchBarProps} from 'sentry/components/smartSearchBar';
+import {MAX_QUERY_LENGTH} from 'sentry/constants';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {FunctionCall, VersionedFunctionCalls} from 'sentry/types/profiling/core';
+import {Container, NumberContainer} from 'sentry/utils/discover/styles';
+import {decodeScalar} from 'sentry/utils/queryString';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+
+interface FunctionsContentProps {
+  error: string | null;
+  isLoading: boolean;
+  version: string;
+  versionedFunctions: VersionedFunctionCalls | null;
+}
+
+function FunctionsContent(props: FunctionsContentProps) {
+  const location = useLocation();
+  const organization = useOrganization();
+  const query = decodeScalar(location.query.query, '');
```

Comment:
Do we need to decode on every rerender or can we just skip this with useMemo()? I think otherwise this will unnecessarily decode every time.